### ISSUE TITLE
Model checkpoints as follow up from callback functions

### ISSUE THREAD

**Issue by CatStark** [jina-ai/finetuner#240]

**Comment by azayz** [comment]
Create ModelcheckpointCallback class which extends BaseCallback and does the following (or gives the possibility to choose from the following) : 

- [ ] save model (keras, pytorch and paddle) on every epoch end.
- [ ] save model (keras, pytorch and paddle) on every batch end. 
- [ ] choose whether to keep only best model.  
- [ ] choose whether to monitor train or validation
- [ ] choose whether to monitor other metrics different than loss

**Comment by tadejsv** [comment]
> save model (keras, pytorch and paddle) on every batch end.

No, not on batch end, only epoch. Otherwise it's too frequent and completely slows down training. End of epoch is a very natural breakpoint and currently we don't have usecases that would call for anything else

**Comment by azayz** [comment]
Yep you are right I thought about it and now I'm thinking about saving after a specific number of batches
